### PR TITLE
rename product to item and separate product and item

### PR DESCRIPTION
**Pull request by sabrina-li**
refactoring productRepo into separate repositories to decouple product and item. Separate ProductRepo into 2 different repositories. 
Product - getting from REST API
Items - used to persist the cart locally

They currently still share the same data model(will need to refactor at some point) this change will make life easier for that too.

architecture diagram updated accordingly

### CODE REVIEW

File: java/app/src/main/java/com/fullstorydev/shoppedemo/data/AppDatabase.java [hunk 1]
```diff
@@ -5,35 +5,42 @@
...
             Executors.newFixedThreadPool(NUMBER_OF_THREADS);
 
-    public static AppDatabase getDatabase(final Context context) {
+    static AppDatabase getDatabase(final Context context) {
         if (INSTANCE == null) {
             synchronized (AppDatabase.class) {
                 if (INSTANCE == null) {
                     INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                             AppDatabase.class, "product_database")
 //                             v1, no need for migration yet, but if you update your schema then you would need to write migrations
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/data/AppDatabase.java:30]: Is this comment still valid?

File: java/app/src/main/java/com/fullstorydev/shoppedemo/ui/market/MarketFragment.java [hunk 2]
```diff
@@ -39,15 +36,10 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
-            @Override
-            public void onChanged(List<Product> products) {
-                mMarketProductAdapter.setProductList(products);
-            }
-        });
+        marketViewModel.getProductList().observe(this.getViewLifecycleOwner(), products -> mMarketProductAdapter.setProductList(products));
     }
 
-    public void onClickAddToCart(Product product){
+    public void onClickAddToCart(Item product){
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/ui/market/MarketFragment.java:42]: Nit: rename `product` -->`item` for consitency

File: java/app/src/main/java/com/fullstorydev/shoppedemo/ui/market/MarketViewModel.java [hunk 3]
```diff
@@ -5,26 +5,27 @@
...
 
-    public LiveData<List<Product>> getProductList() {
+    public LiveData<List<Item>> getProductList() {
         return mProductList;
     }
 
-    public void increaseQuantityInCart(Product product){
-        mRepository.increaseQuantityInCart(product);
-    }
+    public void increaseQuantityInCart(Item product){ mItemRepository.increaseQuantityInCart(product); }
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/ui/market/MarketViewModel.java:30]: Nit: rename `product` -->`item` for consistency

**Comment by patrick-fs** [comment]
> architecture diagram updated accordingly

Hey Sabrina, I didn't see and architecture diagram update. Is this a missing commit?

**Comment by sabrina-li** [comment]
> > architecture diagram updated accordingly
> 
> Hey Sabrina, I didn't see and architecture diagram update. Is this a missing commit?

The binary file is not shown in the change by default, it's part of the PR

**Pull request merged**